lam4-backend/src/Lam4/Render/Render.hs

```diff
@@ -0,0 +1,173 @@
+{-# LANGUAGE QuasiQuotes, OverloadedRecordDot, GADTs #-}
+
+module Lam4.Render.Render where
+
+import qualified Base.Text                as T
+import           Lam4.Expr.CommonSyntax
+import           Lam4.Expr.ConcreteSyntax
+import qualified Lam4.Expr.Name           as N (Name(..), ReferentStatus(..))
+import           Data.String.Interpolate  (i)
+import           Control.Lens             ((&), (%~))
+import           Control.Lens.Regex.Text  (match, regex)
+import           Paths_lam4_backend       (getDataFileName)
+
+-- GF-related stuff
+import           Lam4.Render.Lam4Gf
+import qualified PGF
+
+-- Loosely copied from dsl/…/natural4
+data NLGEnv = NLGEnv
+  { gfGrammar :: PGF.PGF
+  , gfLang :: PGF.Language
+--  , gfParse :: Type -> T.Text -> [Expr]
+  , gfLin :: PGF.Expr -> T.Text
+  }
+
+gfPath :: String -> String
+gfPath x = [i|gf-grammar/#{x}|]
+
+myNLGEnv :: IO NLGEnv
+myNLGEnv = do
+  grammarFile <- getDataFileName $ gfPath "Lam4.pgf"
+  gr <- PGF.readPGF grammarFile
+  let lang = getLang "Lam4Eng" gr
+      -- myParse typ txt = parse gr eng typ (Text.unpack txt)
+      myLin = postprocessText . T.pack . PGF.linearize gr lang
+  pure $ NLGEnv gr lang myLin
+  where
+    postprocessText :: T.Text -> T.Text
+    postprocessText = newlines . tabs . rmBIND
+    rmBIND :: T.Text -> T.Text
+    rmBIND input = input & [regex|\s+&\+\s+|] . match %~ const ""
+
+    tabs :: T.Text -> T.Text
+    tabs = T.map (\c -> if c == '°' then ' ' else c)
```

Comment:
This probably isn't worth changing in the short term, but my preference would be to have some kind of PostprocessConfig record that captures info like what the newline / tab characters / delimiters are and how they should be handled, so that it'll be easier to change  things like that. But maybe not worth changing right now.

Reply:
Sure! Happy to make things less ad hoc later.